core/domain/wipeout_service.py

```diff
@@ -109,11 +107,103 @@ def _delete_user_models(user_id):
 
 
+def _hard_delete_explorations_and_collections(pending_deletion_model):
+    """Hard delete the exploration and collection models that are private and
+    solely owned by the user.
+
+    Args:
+        pending_deletion_model: PendingDeletionRequestModel.
+    """
+    exp_services.delete_explorations(
+        pending_deletion_model.id, pending_deletion_model.exploration_ids)
+    collection_services.delete_collections(
+        pending_deletion_model.id, pending_deletion_model.collection_ids)
+
+
+def _generate_activity_to_user_ids_mapping(activity_ids):
+    """Generate mapping from activity IDs to user IDs.
+
+    Args:
+        activity_ids: list(str). List of activity IDs for which to generate
+        new user IDs.
+
+    Returns:
+        dict(str, str). Mapping between the activity IDs and pseudonymized user
+        IDs.
+    """
+    return {
+        activity_id: user_models.PseudonymizedUserModel.get_new_id()
+        for activity_id in activity_ids
+    }
+
+
+def _delete_story_models(pending_deletion_model):
+    """Pseudonymize the story models for the user with user_id.
+
+    Args:
+        pending_deletion_model: PendingDeletionRequestModel.
+    """
+    user_id = pending_deletion_model.id
+    metadata_models = story_models.StorySnapshotMetadataModel.query(
+        story_models.StorySnapshotMetadataModel.committer_id == user_id
+    ).fetch()
+    story_ids = set([
+        model.get_unversioned_instance_id() for model in metadata_models])
+    commit_log_models = story_models.StoryCommitLogEntryModel.query(
+        story_models.StoryCommitLogEntryModel.user_id == user_id
+    ).fetch()
+    story_ids = story_ids | set(model.story_id for model in commit_log_models)
+
+    if not pending_deletion_model.story_mappings:
+        pending_deletion_model.story_mappings = (
+            _generate_activity_to_user_ids_mapping(story_ids))
+        pending_deletion_model.put()
+
+    def _pseudonymize_models(story_related_models, user_id):
+        """Pseudonymize user ID fields in the models.
+
+        Args:
+            story_related_models: list(ndb.Model). Models which user IDs should
+                be pseudonymized.
+            user_id: str. New pseudonymized user ID to be used for the models.
+        """
+        metadata_models = [
+            model for model in story_related_models
+            if isinstance(model, story_models.StorySnapshotMetadataModel)]
+        for metadata_model in metadata_models:
+            metadata_model.committer_id = user_id
+        story_models.StorySnapshotMetadataModel.put_multi(
+            metadata_models, update_last_updated_time=False)
+
+        commit_log_models = [
+            model for model in story_related_models
+            if isinstance(model, story_models.StoryCommitLogEntryModel)]
+        for commit_log_model in commit_log_models:
+            commit_log_model.user_id = user_id
+            commit_log_model.username = 'ANONYMOUS'
```

Comment:
Does this show up anywhere in the UI? Why is it capitalized? Should it be None?

Also this should be a named constant.